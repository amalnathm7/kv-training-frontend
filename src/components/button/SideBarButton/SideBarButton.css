.item.container {
    font-size: 14px;
    color: white;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    right: 0px; 
    top: 120px;
    width: 95%;
    height: 60px;
    border-top-left-radius: 30px;
    border-bottom-left-radius: 30px;
    background-color: #03AEEE;
    margin: 10px 0; 
}

.icon.container {
    margin-left: 15px;
    margin-right: 15px;
    background-color: white;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 10px;
    border-radius: 100%;
}

