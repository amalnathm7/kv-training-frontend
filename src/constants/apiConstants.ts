export const API_BASE_URL = 'http://localhost:3000';
export const GET_EMPLOYEE_LIST = 'GET_EMPLOYEE_LIST';
export const GET_OPENING_LIST = 'GET_OPENING_LIST';
export const GET_REFERRAL_LIST = 'GET_REFERRAL_LIST';
export const GET_MY_REFERRAL_LIST = 'GET_MY_REFERRAL_LIST';
export const GET_APPLICATION_LIST = 'GET_APPLICATION_LIST';
