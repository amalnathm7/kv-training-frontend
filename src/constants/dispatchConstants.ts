class DispatchConstants {
    static createEmployee = "CREATE_EMPLOYEE";
    static editEmployee = "EDIT_EMPLOYEE";
    static deleteEmployee = "DELETE_EMPLOYEE";
};

export default DispatchConstants;
