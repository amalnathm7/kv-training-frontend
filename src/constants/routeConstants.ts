export class RouteConstants {
    static login = "/login";
    static employee = "/employee";
    static employeeApi = "/employees";
    static departmentApi = "/departments";
    static roleApi = "/api/roles";
    static loginApi = "/employees/login";
};
