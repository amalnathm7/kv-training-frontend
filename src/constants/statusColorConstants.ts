export const StatusColor = {
  Active: '#D3F4BE',
  Probation: '#F5ECB8',
  Inactive: '#FFBFBF',
  Received: '#F5ECB8',
  Review: '#F5ECB8',
  Round_1: '#EAF9FF',
  Round_2: '#EAF9FF',
  Round_3: '#EAF9FF',
  Offered: '#FFBFBF',
  Offer_Accepted: '#FFBFBF',
  Offer_Declined: '#FFBFBF',
  Rejected: '#FFBFBF',
  Hired: '#FFBFBF'
};
