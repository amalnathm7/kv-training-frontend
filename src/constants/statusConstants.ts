export const StatusColor = {
  Active: '#D3F4BE',
  Probation: '#F5ECB8',
  Inactive: '#FFBFBF',
  Received: '#F5ECB8',
  Review: '#F5ECB8',
  Round_1: '#F5ECB8',
  Round_2: '#F5ECB8',
  Round_3: '#F5ECB8',
  Offered: '#D3F4BE',
  Offer_Accepted: '#D3F4BE',
  Offer_Declined: '#FFBFBF',
  Rejected: '#FFBFBF',
  Hired: '#D3F4BE',
  Eligible: '#F5ECB8',
  Processing: '#F5ECB8',
  Approved: '#D3F4BE'
};

export const candidateStatuses = [
  'Received',
  'Review',
  'Round 1',
  'Round 2',
  'Round 3',
  'Offered',
  'Offer Accepted',
  'Offer Declined',
  'Rejected',
  'Hired'
];
