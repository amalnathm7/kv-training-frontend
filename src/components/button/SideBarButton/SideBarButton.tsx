import React from 'react';
import './SideBarButton.css';

export type SideBarButtonPropsType = {
  imgIcon: string;
  headerText: string;
  isSelected: boolean;
  onClick: () => void;
};

const SideBarButton: React.FC<SideBarButtonPropsType> = (props) => {
  return (
    <div
      onClick={props.onClick}
      className={props.isSelected ? 'item container selected' : 'item container'}
      data-testid='sidebar-button-test'
    >
      <div className={props.isSelected ? 'icon container selected' : 'icon container'}>
        <img
          className='filtered'
          style={{ height: '15px', width: '15px' }}
          src={props.imgIcon}
          alt='Icon'
        ></img>
      </div>
      <h3>{props.headerText}</h3>
    </div>
  );
};

export default SideBarButton;
