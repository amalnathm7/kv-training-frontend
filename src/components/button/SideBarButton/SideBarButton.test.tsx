import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import SideBarButton, { SideBarButtonPropsType } from './SideBarButton';
import { RouteConstants } from '../../../constants/routeConstants';
import { BrowserRouter } from 'react-router-dom';

describe('Sidebar Button Props Test', () => {
  test('If sidebarbutton rendered correctly', () => {
    const props: SideBarButtonPropsType = {
      imgIcon: 'icon',
      headerText: 'text',
      route: RouteConstants.employee
    };

    const element = render(
      <BrowserRouter>
        <SideBarButton {...props} />
      </BrowserRouter>
    );

    expect(element).toMatchSnapshot();
  });

  test('If onClick called', () => {
    const props: SideBarButtonPropsType = {
      imgIcon: 'icon',
      headerText: 'text',
      route: RouteConstants.employee
    };

    render(
      <BrowserRouter>
        <SideBarButton {...props} />
      </BrowserRouter>
    );
    const element = screen.getByTestId('sidebar-button-test');

    element.click();
    expect(element).toMatchSnapshot();
  });
});
