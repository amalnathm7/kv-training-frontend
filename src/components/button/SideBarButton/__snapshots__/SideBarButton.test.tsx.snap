// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Button Props Test If onClick called 1`] = `
<div
  class="item container"
  data-testid="sidebar-button-test"
>
  <div
    class="icon container"
  >
    <img
      alt="Icon"
      class="filtered"
      src="icon"
      style="height: 15px; width: 15px;"
    />
  </div>
  <h3>
    text
  </h3>
</div>
`;

exports[`Sidebar Button Props Test If sidebarbutton rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="item container"
        data-testid="sidebar-button-test"
      >
        <div
          class="icon container"
        >
          <img
            alt="Icon"
            class="filtered"
            src="icon"
            style="height: 15px; width: 15px;"
          />
        </div>
        <h3>
          text
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="item container"
      data-testid="sidebar-button-test"
    >
      <div
        class="icon container"
      >
        <img
          alt="Icon"
          class="filtered"
          src="icon"
          style="height: 15px; width: 15px;"
        />
      </div>
      <h3>
        text
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
